lib/source/file.js

```diff
@@ -0,0 +1,167 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+
+class FileParser {
+  constructor() {
+    this.properties = Object.create(null);
+  }
+
+  /**
+   * Parse the file contents
+   * @param {Buffer} data
+   */
+  update(data) {
+    let properties = Object.create(null);
```

Comment:
`Object.create({})` which is what `properties = {}` does under the hood brings along all of the extra stuff in the `Object` prototype.
```javascript
> const foo = Object.create(null);
> const bar = Object.create({});
> typeof foo.hasOwnProperty;
'undefined'
> typeof bar.hasOwnProperty;
'function'
```

I don't think there's a huge amount of difference creating the object from `null` vs having `Object` in the inheritance chain but who knows. It strikes me as a guard against potential side effects, especially when we don't need anything that the `Object` proto provides.